### PR TITLE
add device TICMeter

### PR DESCRIPTION
**Pull request by xmow49**
Add the device "TICMeter" converter and the OTA.

### CODE REVIEW

File: src/devices/gmmts.ts [hunk 1]
```diff
@@ -0,0 +1,848 @@
+import {Definition, Fz, Tz, KeyValue, Zh} from '../lib/types';
+/* eslint-disable camelcase */
+/* eslint-disable max-len */
```
Comment (Koenkk) [inline src/devices/gmmts.ts:3]: Please don't do any `eslint-disable` here

File: src/devices/gmmts.ts [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+    STD: 'STANDARD',
+};
+
+const mode_tic_enum = ['HISTORIQUE', 'STANDARD', 'AUTO'];
+const mode_elec_enum = ['MONOPHASE', 'TRIPHASE', 'AUTO'];
+const mode_contract_enum = ['AUTO', 'BASE', 'HCHP', 'EJP', 'TEMPO', 'PRODUCTEUR'];
+
+// prettier-ignore
+const ticmeter_options = [
+    e.numeric(`refresh_rate`, ea.SET).withValueMin(60).withDescription(`Temps d'actualisation des valeurs statiques (celles qui possèdent des boutons refresh). Par défaut: ${DEFAULT_POLL_INTERVAL} s`).withValueMin(60).withValueMax(3600),
```
Comment (Koenkk) [inline src/devices/gmmts.ts:61]: There is a lot of French here, please use English instead.

File: src/devices/gmmts.ts [hunk 3]
```diff
@@ -0,0 +1,848 @@
+import {Definition, Fz, Tz, KeyValue, Zh} from '../lib/types';
```
Comment (Koenkk) [inline src/devices/gmmts.ts:1]: This is quite some code to support just a single device, would it be possible to simply somehow by relying on standard functionality like reporting?

Comment (xmow49) [reply src/devices/gmmts.ts:1]: This device is a smart energy monitor, witch is pluged to a electric meter and read data.
Depending of the electricity contract, commincation mode, number of phase, eletricity producer or not, entitys appears or not.
I already use reporting for somes attributes.
I use custom fz for haElectricalMeasurement beacause the device dont have all attributes available.

I dont know how to reduce the code.

Thanks

File: src/devices/gmmts.ts (hunk 2 above)
Comment (xmow49) [reply src/devices/gmmts.ts:61]: This product is intended only for the French market (French electric meter). Is it possible to add translations?

File: src/devices/gmmts.ts (hunk 1 above)
Comment (xmow49) [reply src/devices/gmmts.ts:3]: Where I can use it ?

Comment (Koenkk) [reply src/devices/gmmts.ts:3]: Just remove it

File: src/devices/gmmts.ts (hunk 2 above)
Comment (Koenkk) [reply src/devices/gmmts.ts:61]: Translations are not support for this, but please translate it to English (ZLinky_TIC is also in English)

File: src/devices/gmmts.ts (hunk 1 above)
Comment (xmow49) [reply src/devices/gmmts.ts:3]: Hi,
If I remove them, I have a lot of lint errors:
image

If I remove spaces; and cut the line at 150 characters; the code becomes very ugly.
Can I keep at least 
/* eslint-disable max-len */
/* eslint-disable no-multi-spaces */ 
?
Thanks

Comment (Koenkk) [reply src/devices/gmmts.ts:3]: Then let's keep those

File: src/devices/gmmts.ts (hunk 2 above)
Comment (xmow49) [reply src/devices/gmmts.ts]: I added an option to change the language.

File: src/devices/gmmts.ts (hunk 3 above)
Comment (xmow49) [reply src/devices/gmmts.ts:1]: I edited some fzlocal to reduce the code size.

**Pull request merged**